modules/context/package.go

```diff
@@ -79,7 +80,7 @@ func packageAssignment(ctx *Context, errCb func(int, string, interface{})) {
 			}
 		}
 		// 2. If authorize level is none, check if org is visible to user
-		if ctx.Package.AccessMode == perm.AccessModeNone && organization.HasOrgOrUserVisible(ctx, ctx.Package.Owner, ctx.Doer) {
+		if ctx.Package.AccessMode == perm.AccessModeNone && organization.HasOrgOrUserVisible(ctx, ctx.Package.Owner, ctx.Doer) && !setting.Service.RequireSignInView {
 			ctx.Package.AccessMode = perm.AccessModeRead
 		}
 	} else {
```

Comment:
Why not on this branch?

Reply:
You talk about line 85? The check should be on every `ctx.Doer == nil` branch.

Reply:
Hmm if the user is nil and RequireSignInView we should be AccessNone and have no access at all. Perhaps this should just be at the start?